src/qibolab/instruments/emulator/pulse_simulator.py

```diff
@@ -466,22 +465,31 @@ def channel_translator(platform_channel_name, frequency):
                     # need to first set pulse._if in GHz to use modulated_waveform_i method
                     pulse._if = pulse.frequency / 1e9
 
-                    pulse_signal = pulse.modulated_waveform_i(
-                        sim_sampling_boost
-                    ).data  # *np.sqrt(2)
-                    end = start + len(pulse_signal) / sim_sampling_boost
+                    i_env = pulse.envelope_waveform_i(sim_sampling_boost).data
+                    q_env = pulse.envelope_waveform_q(sim_sampling_boost).data
+
+                    # Qubit drive microwave signals
+                    end = start + len(i_env) / sim_sampling_boost
                     t = (
                         np.arange(start * sim_sampling_boost, end * sim_sampling_boost)
                         / sampling_rate
                         / sim_sampling_boost
                     )
+                    cosalpha = np.cos(
+                        2 * np.pi * pulse._if * sampling_rate * t + pulse.relative_phase
+                    )
+                    sinalpha = np.sin(
+                        2 * np.pi * pulse._if * sampling_rate * t + pulse.relative_phase
+                    )
+                    pulse_signal = i_env * sinalpha + q_env * cosalpha
```

Comment:
i_env/q_env and sinalpha/cosalpha can have one-off different sizes due to rounding errors for `t`, so this multiplication can fail.

Reply:
But this is what you are using also in icarusqfpga.py, so it will be consistent with the waveforms you generate for your hardware... or not?

Reply:
```
  File "C:\Users\pault\Documents\repos\qibolab\src\qibolab\platform.py", line 170, in _execute
    new_result = instrument.play(
                 ^^^^^^^^^^^^^^^^
  File "C:\Users\pault\Documents\repos\qibolab\src\qibolab\instruments\emulator\pulse_simulator.py", line 160, in play
    self.run_pulse_simulation(sequence, self.instant_measurement)
  File "C:\Users\pault\Documents\repos\qibolab\src\qibolab\instruments\emulator\pulse_simulator.py", line 118, in run_pulse_simulation
    channel_waveforms = ps_to_waveform_dict(
                        ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pault\Documents\repos\qibolab\src\qibolab\instruments\emulator\pulse_simulator.py", line 484, in ps_to_waveform_dict
    pulse_signal = i_env * sinalpha + q_env * cosalpha
                   ~~~~~~^~~~~~~~~~
ValueError: operands could not be broadcast together with shapes (611,) (612,)
```
For IcarusQ, my `start` and `end` values are integers that represent the number of samples, but at L474, they are floats corresponding to the actual starting/ending times, so there could be some rounding errors with `np.arange`, as I have encountered.

Reply:
Setting L474
```python
np.arange(int(start * sim_sampling_boost), int(end * sim_sampling_boost))
```
seems to have fixed the issue

Reply:
Fixed with https://github.com/qiboteam/qibolab/pull/849/commits/507b5f6e2a5d9f2d56670b15b6f031bcab64cbfb